spec/models/accounting/transaction_spec.rb

```diff
@@ -76,4 +76,70 @@
       expect(subject.project_id).to eq loan.id
     end
   end
+
+  context 'when disbursement transaction' do
+    let(:transaction_type) { 'disbursement' }
+
+    context 'without qb_id' do
+      it 'can save without amount' do
+        params = {
+          amount: nil,
+          txn_date: '2017-10-31',
+          private_note: 'a memo',
+          description: 'desc',
+          project_id: loan.id,
+          qb_transaction_type: transaction_type,
+          qb_id: nil
+        }
+        expect { create(:accounting_transaction, params) }.to raise_error(ActiveRecord::RecordInvalid)
+      end
+    end
+    context 'with qb_id' do
+      it 'requires an amount to save' do
+        params = {
+          amount: nil,
+          txn_date: '2017-10-31',
+          private_note: 'a memo',
+          description: 'desc',
+          project_id: loan.id,
+          qb_transaction_type: transaction_type,
+          qb_id: 123
+        }
+        expect { create(:accounting_transaction, params) }.to raise_error(ActiveRecord::RecordInvalid)
+      end
+    end
+  end
+
+  context 'when interest transaction' do
+    let(:transaction_type) { 'interest' }
+
+    context 'without qb_id' do
+      it 'can save without amount' do
+        params = {
+          amount: nil,
+          txn_date: '2017-10-31',
+          private_note: 'a memo',
+          description: 'desc',
+          project_id: loan.id,
+          qb_transaction_type: transaction_type,
+          qb_id: nil
+        }
+        expect { create(:accounting_transaction, params) }.not_to raise_error
+      end
+    end
+    context 'with qb_id' do
+      it 'requires an amount to save' do
+        params = {
+          amount: nil,
+          txn_date: '2017-10-31',
+          private_note: 'a memo',
+          description: 'desc',
+          project_id: loan.id,
+          qb_transaction_type: transaction_type,
+          qb_id: 123
+        }
+        expect { create(:accounting_transaction, params) }.to raise_error(ActiveRecord::RecordInvalid)
+      end
+    end
```

Comment:
Can you please DRY up these specs. Most of them are identical attributes. Those can go in a let block at the top and you can just do `merge` to set the ones you're testing.

This will make it much clearer what you're actually changing from assertion to assertion.

In general if you could please make an effort to keep specs DRY. You're good about this with regular code but often your specs are much longer than they need to be.

Thanks!